server/model/monitor.js

```diff
@@ -1018,18 +1016,35 @@ class Monitor extends BeanModel {
             }
 
             return res;
-        } catch (e) {
+        } catch (error) {
+
+            /**
+             * Make a single attempt to obtain an new access token in the event that
+             * the recent api request failed for authentication purposes
+             */
+            if (this.auth_method === "oauth2-cc" && error.response.status === 401 && !options.isReattempt) {
+                this.oauthAccessToken = await this.makeOidcTokenClientCredentialsRequest();
```

Comment:
For information purposes I would add a debug log line stating that there will be an internal retry with a new token. But it is not really needed.

Reply:
This occurs in the first line of the makeOidcTokenClientCredentialsRequest method